src/libluau.zig

```diff
@@ -927,6 +947,22 @@ pub const Lua = struct {
         return error.Fail;
     }
 
+    /// Converts the Lua value at the given `index` to a 3- or 4-vector.
+    /// The Lua value must be a vector.
+    pub fn toVector(lua: *Lua, index: i32) ![luau_vector_size]f32 {
+        const res = c.lua_tovector(lua.state, index);
+        if (res) |r| {
+            if (luau_vector_size == 3) {
+                return [_]f32{ r[0], r[1], r[2] };
+            } else if (luau_vector_size == 4) {
+                return [_]f32{ r[0], r[1], r[2], r[3] };
+            } else {
+                @compileError("luau_vector_size must be 3 or 4");
```

Comment:
I think this compile error can be removed now

Reply:
I changed this to 

```
            switch (luau_vector_size) {
                3 => return [_]f32{ r[0], r[1], r[2] },
                4 => return [_]f32{ r[0], r[1], r[2], r[3] },
                else => @compileError("invalid luau_vector size - should not happen"),
            }
```

IMO it reads better with the `@compileError` as that documents better that only 3 or 4 are legal here.